lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -2967,8 +2967,8 @@ def _set_disk_features(
         allowed_types = azure_runbook.image.disk_controller_type
         if node_space.disk.disk_controller_type:
             allowed_types = search_space.intersect_setspace_by_priority(
-                node_space.disk.disk_controller_type,  # type: ignore
                 azure_runbook.image.disk_controller_type,  # type: ignore
+                node_space.disk.disk_controller_type,  # type: ignore
```

Comment:
Why was their order changed? Does it make a difference?

Reply:
Yes, swapped them so that node_space is the capability and azure_runbook is the requirement.
The merge operation enforces that capability is not None, but requirement is allowed to be none.

https://github.com/microsoft/lisa/blob/f2c6ee1329a227ab9ad14da30c2d31d9abc7edd5/lisa/search_space.py#L499-L500